android/src/main/kotlin/com/apptentive/apptentive_flutter/ApptentiveFlutterPlugin.kt

```diff
@@ -286,74 +302,108 @@ class ApptentiveFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware
   // Register listeners for native callbacks:
   private fun registerListeners(result: Result) {
     try {
-      Apptentive.setOnSurveyFinishedListener { channel.invokeMethod("onSurveyFinished", mapOf("completed" to it)) }
-      Apptentive.addUnreadMessagesListener { channel.invokeMethod("onUnreadMessageCountChanged", mapOf("count" to it)) }
+      registerForSurveyListener()
+      registerForUnreadMessageListener()
       result.success(true)
     } catch(e: Exception) {
       result.error(ERROR_CODE, "Failed to register Apptentive listeners.", e.toString())
     }
   }
 
+  private fun registerForSurveyListener() {
+    Apptentive.eventNotificationObservable.observe { notification ->
+      val name = notification?.name
+      val interaction = notification?.interaction
+      val vendor = notification?.vendor
+      val interactionId = notification?.interactionId
+      val notificationText = "Name: \"$name\". Vendor: \"$vendor\". " +
+              "Interaction: \"$interaction\". Interaction ID: $interactionId"
+      Log.d(LogTags.EVENT_NOTIFICATION, notificationText)
```

Comment:
It would be better to include this in the native SDK instead

Reply:
It would be good to have this until we have it in the native.